### PR TITLE
Check-in translation PoC

### PR DESCRIPTION
**Pull request by acrollet**
## Description

The check-in experience group has been asked to translate the check-in and pre-check-in apps by May 2. We have a working PoC for the whole app, but this PR provides the functionality to just the error page. The purpose is to provide a more streamlined PR for the frontend review group to review and request the required dependencies:

- [i18next](https://www.i18next.com)
- [react-i18next](https://react.i18next.com)
- [i18next-browser-languagedetector](https://github.com/i18next/i18next-browser-languageDetector)
- [i18next-resources-to-backend](https://github.com/i18next/i18next-resources-to-backend)

react-i18next was chosen because of its adoption in the React community, and due to its [recommendation in past platform research](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/content-localization/development/Decision-Log-I18N-JS-Libraries.md)

To test locally:

- Start the mock API: `yarn mock-api --responses src/applications/check-in/api/local-mock-api/index.js`
- Run the app: `yarn watch --env entry=check-in`
- Go to the error page: http://localhost:3001/health-care/appointment-check-in/error

You should see the app in english with English / Español buttons at the bottom. If you visit with your browser language set to Spanish, it should automatically select Spanish as the language.

NB: the translations in the file are machine translated for the PoC, but translation capability is behind a feature flag and will not be released to the public until official translations have been obtained & reviewed.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#38309

## Testing done

manual testing, updated unit and e2e tests

## Screenshots

Screen Shot 2022-03-21 at 4 08 58 PM

### CODE REVIEW

File: src/applications/check-in/components/LanguagePicker.jsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import withTranslationEnabled from '../containers/withTranslationEnabled';
+
+function LanguagePicker() {
+  const { i18n } = useTranslation();
+
+  function changeLanguage(e) {
+    i18n.changeLanguage(e.target.value);
+  }
+
+  function buttonClass(language) {
```
Comment (dcgoodwin2112) [inline src/applications/check-in/components/LanguagePicker.jsx:13]: We could potentially make this "if elseif else" block a switch statement if we anticipate adding several more lanaguages in the future. It's also perfectly fine as is though.

File: src/applications/check-in/utils/i18n/i18n.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import i18n from 'i18next';
+import { initReactI18next } from 'react-i18next';
+import LanguageDetector from 'i18next-browser-languagedetector';
+// import resourcesToBackend from 'i18next-resources-to-backend';
+import en from '../../locales/en/translation.json';
+
+i18n
+  /*
+  .use(
```
Comment (dcgoodwin2112) [inline src/applications/check-in/utils/i18n/i18n.js:9]: Can the commented out code block be removed or do we still need this one?

**Review by dcgoodwin2112 — APPROVED** [review]
I added a couple very minor comments, but this looks great!

File: src/applications/check-in/utils/i18n/i18n.js (hunk 2 above)
Comment (dcgoodwin2112) [reply src/applications/check-in/utils/i18n/i18n.js:9]: Duplicate comment - marking resolved

File: src/applications/check-in/components/LanguagePicker.jsx (hunk 1 above)
Comment (dcgoodwin2112) [reply src/applications/check-in/components/LanguagePicker.jsx:13]: Duplicate comment -- marking resolved

**Review by mdewey — APPROVED** [review]
LGTM

File: src/applications/check-in/day-of/pages/Error.jsx [hunk 3]
```diff
@@ -20,6 +25,7 @@ const Error = () => {
       )}
       <Footer />
       <BackToHome />
+      <LanguagePicker />
```
[the remark this answers is not in the document]
Comment (acrollet) [reply src/applications/check-in/day-of/pages/Error.jsx:28]: great idea, I created #38881

**Review by theodur — APPROVED** [review]

**Pull request merged**